catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js

```diff
@@ -137,7 +143,27 @@ define([
             template: 'sourceList',
             itemView: SourceView.SourceRow,
             emptyView: EmptyView.sources,
-            itemViewContainer: 'tbody'
+            itemViewContainer: 'tbody',
+
+            initialize: function() {
+                this.hidePageLoading();
+                this.showSourcesLoading();
+            },
+            collectionEvents: {
+                "request": "showSourcesLoading",
+                "sync": "hideSourcesLoading"
+            },
+            showSourcesLoading: function() {
+                this.$el.addClass('hide');
+                $("#sources-loading").removeClass('hide');
+            },
+            hideSourcesLoading: function() {
+                  $("#sources-loading").addClass('hide');
```

Comment:
If you are not able to make this model driven, at least make the loading element part of this template and then use `toggleClass`.

https://api.jquery.com/toggleClass/